packages/store/src/decorators/state.ts

```diff
@@ -1,42 +1,28 @@
-import { ensureStoreMetadata } from '../internal/internals';
-import { StoreOptions, META_KEY } from '../symbols';
-
-const stateNameRegex = new RegExp('^[a-zA-Z0-9_]+$');
-
-/**
- * Error message
- * @ignore
- */
-export const stateNameErrorMessage = (name: string) =>
-  `${name} is not a valid state name. It needs to be a valid object property name.`;
+import { ensureStoreMetadata, MetaDataModel, StateClass } from '../internal/internals';
+import { META_KEY, META_OPTIONS_KEY, StoreOptions } from '../symbols';
+import { StoreValidators } from '../utils/store-validators';
 
 /**
  * Decorates a class with ngxs state information.
  */
 export function State<T>(options: StoreOptions<T>) {
-  return function(target: any) {
-    const meta = ensureStoreMetadata(target);
+  return function(target: StateClass) {
+    const meta: MetaDataModel = ensureStoreMetadata(target);
+    const targetReference: StateClass = Object.getPrototypeOf(target);
+    const inheritanceOptions: Partial<StoreOptions<T>> = target[META_OPTIONS_KEY] || {};
```

Comment:
```suggestion
    const inheritanceOptions: Partial<StoreOptions<T>> = targetReference[META_OPTIONS_KEY] || {};
```
Surely we want the options from the prototype?
Also maybe `inheritedStateClass` might be a better name than `targetReference`?